solver/src/driver.rs

```diff
@@ -156,9 +159,19 @@ impl Driver {
         }
     }
 
-    async fn can_settle(&self, settlement: RatedSettlement) -> Result<bool> {
+    async fn can_settle_without_liquidity(&self, settlement: &RatedSettlement) -> Result<bool> {
+        // We don't want to buy tokens that we don't trust. If no list is set, we settle with external liquidity.
+        if !self
+            .market_makable_token_list
+            .as_ref()
+            .map(|list| is_only_selling_trusted_tokens(settlement.clone().into(), list))
+            .unwrap_or(false)
+        {
+            return Ok(false);
+        }
+
```

Comment:
Just checking, when we can't get the trusted token list then we never use buffers, right?

Reply:
Yes, this is to be more "safe" and have this feature opt-in.